src/main/java/net/snowflake/client/jdbc/diagnostic/CertificateDiagnosticCheck.java

```diff
@@ -0,0 +1,70 @@
+package net.snowflake.client.jdbc.diagnostic;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.Proxy;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import net.snowflake.client.log.SFLogger;
+import net.snowflake.client.log.SFLoggerFactory;
+
+public class CertificateDiagnosticCheck extends DiagnosticCheck {
+
+  private static final String SECURE_SOCKET_PROTOCOL = "TLS";
+
+  private static final SFLogger logger =
+      SFLoggerFactory.getLogger(CertificateDiagnosticCheck.class);
+
+  public CertificateDiagnosticCheck(ProxyConfig proxyConfig) {
+    super("SSL/TLS Certificate Test", proxyConfig);
+  }
+
+  @Override
+  public void run(SnowflakeEndpoint snowflakeEndpoint) {
+    super.run(snowflakeEndpoint);
+    if (snowflakeEndpoint.isSslEnabled()) {
+      String hostname = snowflakeEndpoint.getHost();
+      String urlString = "https://" + hostname + ":" + snowflakeEndpoint.getPort();
+      try {
+        SSLContext sslContext = SSLContext.getInstance(SECURE_SOCKET_PROTOCOL);
+        sslContext.init(null, new TrustManager[] {new DiagnosticTrustManager()}, null);
+        HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.getSocketFactory());
+        Proxy proxy = this.proxyConf.getProxy(snowflakeEndpoint);
+        new URL(urlString).openConnection(proxy).connect();
```

Comment:
Maybe we can have some message confirming it succeeded? Ignore this, if this is logged on another level.

Reply:
HttpAndHttpsDiagnosticCheck uses HttpsURLConnection which again performs a TLS handshake so we there we report if we succeeded or if we failed. However, the limitation there is that if you fail then you have no way of getting information about the server's certificate, which is why we needed this code to create a new SSLContext with a custom TrustManager. The DiagnosticTrustManager just prints out the certificate details but we don't perform any validations or make any conclusions about trusting the server.